src/clientserver.c

```diff
@@ -511,8 +601,25 @@ cmdsrv_main(
 	    // Win32 always works?
 	    res = serverGetVimNames();
 # else
-	    if (xterm_dpy != NULL)
+#  ifdef FEAT_SOCKETSERVER
+	    if (clientserver_method == CLIENTSERVER_METHOD_SOCKET)
+	    {
+		int was_init = socket_server_valid();
```

Comment:
The indentation used in this PR is not following the Vim indentation convention.  Did you indent the code using only tabs?